tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -407,32 +301,171 @@ private static Span StartMvcCoreSpan(Tracer tracer, Span parentSpan, BeforeActio
             return span;
         }
 
-        private void OnHostingHttpRequestInStart(object arg)
+#if NETCOREAPP
+        private void OnNextSlow(string eventName, object arg)
         {
-            var tracer = CurrentTracer;
-            var security = CurrentSecurity;
+            var lastChar = eventName[^1];
+            if (lastChar == 't')
+            {
+                if ((_eventNameCacheFlags & 1) == 0 && eventName.AsSpan().Slice(PrefixLength) is "Hosting.HttpRequestIn.Start")
```

Comment:
Just so I understand, `_eventNameCacheFlags` is only useful in the scenario where the diagnostic observer raises an event that we _don't_ handle, right?

Reply:
yes, there's a log of events we don't handle and I want to avoid string comparisons in each requests.